stackslib/src/config/mod.rs

```diff
@@ -4047,9 +4153,101 @@ impl AtlasConfigFile {
+    ///   a specific contract event key, or a specific asset identifier key).
+    /// - `"stackerdb"`: Subscribes to StackerDB chunk update events.
+    ///   Payloads are sent to the `/stackerdb_chunks` path.
+    /// - `"block_proposal"`: Subscribes to block proposal response events (for Nakamoto consensus).
+    ///   Payloads are sent to the `/proposal_response` path.
+    /// - Smart Contract Event (format: `"{contract_address}.{contract_name}::{event_name}"`):
+    ///   Subscribes to a specific smart contract event. The `{contract_address}.{contract_name}` part identifies the contract,
+    ///   and `::{event_name}` specifies the event defined within that contract.
+    ///   Example: `"ST0000000000000000000000000000000000000000.my-contract::my-custom-event"`
+    ///   These are delivered as part of `/new_block` or `/new_microblocks` payloads, with the "events" array filtered for this specific event.
+    /// - Asset Identifier for FT/NFT Events (format: `"{contract_address}.{contract_name}.{asset_name}"`):
+    ///   Subscribes to events (mint, burn, transfer) for a specific Fungible Token (FT) or Non-Fungible Token (NFT).
+    ///   Example for an FT: `"ST0000000000000000000000000000000000000000.my-ft-contract.my-fungible-token"`
+    ///   These are delivered as part of `/new_block` or `/new_microblocks` payloads, with the "events" array filtered for the specified asset.
+    ///
+    /// **Example `events_keys` in TOML:**
+    /// ```toml
+    /// events_keys = [
+    ///   "burn_blocks",
+    ///   "memtx",
+    ///   "ST0000000000000000000000000000000000000000.my-contract::my-custom-event",  // Smart contract event
+    ///   "ST0000000000000000000000000000000000000000.token-contract.my-ft"         // Fungible token asset event
+    /// ]
+    /// ```
+    ///
+    /// Default: No default. This field is required.
     pub events_keys: Vec<String>,
+    /// Maximum duration (in milliseconds) to wait for the observer endpoint to respond.
+    ///
+    /// When the node sends an event notification to this observer, it will wait at most this long
+    /// for a successful HTTP response (status code 200) before considering the request timed out.
+    /// If a timeout occurs and retries are enabled (see [`EventObserverConfig::disable_retries`]),
+    /// the request will be attempted again according to the retry strategy.
+    ///
+    /// Default: `1_000` ms (1 second).
     pub timeout_ms: Option<u64>,
+    /// Controls whether the node should retry sending event notifications if delivery fails or times out.
+    ///
+    /// - If `false` (default): The node will attempt to deliver event notifications persistently.
+    ///   If an attempt fails (due to network error, timeout, or a non-200 HTTP response),
+    ///   the event payload is typically saved to a local database (if [`NodeConfig::working_dir`] is configured for the node)
```

Comment:
nit, but wouldn't this `(if [`NodeConfig::working_dir`] is configured for the node)` also be true if the key is not configured? the path would be less clear though: `/tmp/stacks-node-xxxxx` but it should still exist (not to mention a restart would start a new genesis sync).